### ISSUE TITLE
Rebuild the output filter

### ISSUE THREAD

**Issue by billerickson** [billerickson/display-posts-shortcode#75]
This plugin was much simpler when initially developed, and the output filter reflected that. It now contains 9 (!) parameters, and is even missing some like the author field. 

I've had quite a few people post issues where things aren't working because they try to write two functions that modify the output, so only the last function does anything since we're returning an entire $output, not just the parts modified.

I'd like to see if there's a way I can refactor this so it's simpler, easier to modify, and of course backwards compatible. 

Other relevant tickets: #66

**Comment by Cohen611** [comment]
Hi Bill,

Love your plugin! How would I fillter for it to show the current's post on the current author that is signed in on their page at the time?

Thanks,
Candace

**Comment by ammist** [comment]
Thanks for directing me here, Bill!

**Comment by billerickson** [comment]
I'd like to deprecate the current output filter and make a new one. The `$output` would be an array of pieces. Something like this:

```php
$output = array(
	array(
		'type' => 'image',
		'content' => '<a class="image" href="http://sample.com"><img src="http://sample.com/image.jpg" /></a>',
	),
	array(
		'type' => 'title',
		'content' => '<a class="title" href="http://sample.com">Sample title</a>',
	),
	array(
		'type' => 'excerpt',
		'content' => '<span class="excerpt-dash">-</span> <span class="excerpt">This is the excerpt</span>',
	)
);
```

Then when filtering it, you can easily target the specific piece you want to change. You can also re-order the elements in the array without affecting other plugins/code that modifies the individual piece.

**Comment by ammist** [comment]
I will happily test this for you.

**Comment by antonyjsmith** [comment]
Seems to be an issue when using the filter it ignores the display_category parameter.

**Comment by billerickson** [comment]
I'm closing this as it will be addressed in Display Posts Shortcode 3.0, a complete rebuild as a Gutenberg block.
